### PR TITLE
Update tools, help, and scope some global vars in Makefile.

### PR DESCRIPTION
**Pull request by jeremyje**
Various fixes to the makefile.
* Ignore delete cluster errors.
* Scope single use global variables.
* Update `make help` command.
* Update versions of tools.
* Reduce the size of the GKE cluster (cheaper and faster startup).
* Fix generation of `install/yaml/` files.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -184,12 +182,13 @@ endif
 help:
 	@cat Makefile | grep ^\#\# | grep -v ^\#\#\# |cut -c 4-
 
+local-cloud-build: LOCAL_CLOUD_BUILD_PUSH = # --push
```
Comment (sawagh) [inline Makefile]: two same tags now? 'local-cloud-build:'

File: Makefile [hunk 2]
```diff
@@ -549,24 +570,31 @@ auth-docker: gcloud docker
...
+create-gke-cluster: GKE_CLUSTER_SIZE_FLAGS = --machine-type n1-standard-1 --num-nodes 1 --max-nodes 3 --disk-size 50
+create-gke-cluster: GKE_FUTURE_COMPAT_FLAGS = --no-enable-basic-auth --no-issue-client-certificate --enable-ip-alias --metadata disable-legacy-endpoints=true --enable-autoupgrade
 create-gke-cluster: build/toolchain/bin/kubectl$(EXE_EXTENSION) gcloud
-	gcloud $(GCP_PROJECT_FLAG) container clusters create $(GKE_CLUSTER_NAME) $(GCP_LOCATION_FLAG) --machine-type n1-standard-4 --tags open-match $(KUBERNETES_COMPAT)
+	gcloud beta $(GCP_PROJECT_FLAG) container clusters create $(GKE_CLUSTER_NAME) $(GCP_LOCATION_FLAG) --cluster-version $(GKE_VERSION) --tags open-match $(GKE_CLUSTER_SIZE_FLAGS) $(GKE_FUTURE_COMPAT_FLAGS)
 	$(KUBECTL) create clusterrolebinding myname-cluster-admin-binding --clusterrole=cluster-admin --user=$(GCLOUD_ACCOUNT_EMAIL)
 
 delete-gke-cluster: gcloud
-	gcloud $(GCP_PROJECT_FLAG) container clusters delete $(GKE_CLUSTER_NAME) $(GCP_LOCATION_FLAG) --quiet
+	-gcloud $(GCP_PROJECT_FLAG) container clusters delete $(GKE_CLUSTER_NAME) $(GCP_LOCATION_FLAG) --quiet
```
Comment (sawagh) [inline Makefile]: what does the '-' here do?

File: Makefile [hunk 3]
```diff
@@ -549,24 +570,31 @@ auth-docker: gcloud docker
...
 
 delete-kind-cluster: build/toolchain/bin/kind$(EXE_EXTENSION) build/toolchain/bin/kubectl$(EXE_EXTENSION)
 	-$(KIND) delete cluster
 
+create-gke-cluster: GKE_VERSION = 1.12.7-gke.17
+create-gke-cluster: GKE_CLUSTER_SIZE_FLAGS = --machine-type n1-standard-1 --num-nodes 1 --max-nodes 3 --disk-size 50
+create-gke-cluster: GKE_FUTURE_COMPAT_FLAGS = --no-enable-basic-auth --no-issue-client-certificate --enable-ip-alias --metadata disable-legacy-endpoints=true --enable-autoupgrade
 create-gke-cluster: build/toolchain/bin/kubectl$(EXE_EXTENSION) gcloud
-	gcloud $(GCP_PROJECT_FLAG) container clusters create $(GKE_CLUSTER_NAME) $(GCP_LOCATION_FLAG) --machine-type n1-standard-4 --tags open-match $(KUBERNETES_COMPAT)
+	gcloud beta $(GCP_PROJECT_FLAG) container clusters create $(GKE_CLUSTER_NAME) $(GCP_LOCATION_FLAG) --cluster-version $(GKE_VERSION) --tags open-match $(GKE_CLUSTER_SIZE_FLAGS) $(GKE_FUTURE_COMPAT_FLAGS)
```
Comment (sawagh) [inline Makefile]: Why are we using gcloud beta now?

File: Makefile [hunk 4]
```diff
@@ -340,10 +344,26 @@ install/yaml/04-grafana-chart.yaml: build/toolchain/bin/helm$(EXE_EXTENSION)
 		--set openmatch.backendapi.install=false \
```
Comment (yfei1) [inline Makefile]: openmatch.backend.install

File: Makefile [hunk 5]
```diff
@@ -340,10 +344,26 @@ install/yaml/04-grafana-chart.yaml: build/toolchain/bin/helm$(EXE_EXTENSION)
...
 		--set grafana.enabled=true \
+		--set jaeger.enabled=false \
 		install/helm/open-match > install/yaml/04-grafana-chart.yaml
 
+install/yaml/05-jaeger-chart.yaml: build/toolchain/bin/helm$(EXE_EXTENSION)
+	mkdir -p install/yaml/
+	$(HELM) template --name $(OPEN_MATCH_CHART_NAME) --namespace $(OPEN_MATCH_KUBERNETES_NAMESPACE) \
+		--set redis.fullnameOverride='$(REDIS_NAME)' \
+		--set openmatch.config.install=false \
+		--set openmatch.backendapi.install=false \
```
Comment (yfei1) [inline Makefile]: ditto

File: Makefile (hunk 3 above)
Comment (jeremyje) [reply Makefile]: These features are only available in the beta API.

File: Makefile (hunk 2 above)
Comment (jeremyje) [reply Makefile]: Make will abort if the error code is not 0. The `-` tells Make to ignore errors.

**Pull request merged**